connect/mirror/src/test/java/org/apache/kafka/connect/mirror/OffsetSyncStoreTest.java

```diff
@@ -30,7 +30,7 @@ public class OffsetSyncStoreTest {
     static class FakeOffsetSyncStore extends OffsetSyncStore {
 
         FakeOffsetSyncStore() {
-            super(null, null);
+            super();
```

Comment:
A few comments:

1. Some cases in this test suite and the `MirrorCheckpointTaskTest` suite are failing because the store isn't started. We might upgrade the visibility of the `OffsetSyncStore.readToEnd` field to `protected`, and expose a method in this class to set the value for that field.
2. Can we get some additional coverage in either/both of the above-mentioned test suites to make sure that a store that hasn't been started yet, or that takes a while to start, results in the expected lack of checkpoints (or translated downstream offsets)?
3. The assertion messages for this test case need to be updated to match the values they're referring to (or, if you prefer, we can just remove specific values from the messages altogether since they can be easily deduced from the source code and line number if an assertion fails)
4. For the `testOffsetTranslation` case, we might consider performing comparisons between two `Optional` instances instead of unconditionally invoking `getAsLong` on the translated downstream offset. For example, `assertEquals(OptionalLong.of(201L), store.translateDownstream(tp, 150), "Failed to translate downstream offsets");`

Reply:
1. I've added a null-check to the real start() method which allows us to call the real start method on the fake offset store.
2. Added
3. I removed the assertion messages as they were just visual noise, the comments seemed easy enough to interpret.
4. Done